analysis.go

```diff
@@ -161,92 +100,80 @@ func checkProjectTask(checkFactory getProjectCheckFactory, JSONResourceFactory D
-		if err != nil {
-			return err
-		}
-		return json.Unmarshal(contents, dest)
+// load loads JSON resource from a given path into dest.
+func load(path string, dest interface{}) error {
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
 	}
+	return json.Unmarshal(contents, dest)
 }
 
 // CheckForWaitingPods checks all pods for any containers in waiting status.
-func CheckForWaitingPods(JSONResourceFactory DumpedJSONResourceFactory) (Result, error) {
-	result := Result{Status: analysisErrorNotDiscovered, StatusMessage: "this issue was not detected", CheckName: "check pods for 'waiting' containers", Info: []Info{}, Events: []Event{}}
-	pods := Pods{}
-	if err := JSONResourceFactory(filepath.Join("definitions", "pods.json"), &pods); err != nil {
-		result.Status = analysisErrorReadingDumpedResource
-		result.StatusMessage = "Error executing task: " + err.Error()
-		return result, err
+func CheckForWaitingPods(pods types.PodList) Result {
+	result := Result{
+		CheckName:     "check pods for 'waiting' containers",
+		Status:        analysisErrorNotDiscovered,
+		StatusMessage: "this issue was not detected",
 	}
-
 	for _, pod := range pods.Items {
 		for _, container := range pod.Status.ContainerStatuses {
 			if container.State.Waiting != nil {
 				result.Status = analysisErrorDiscoveredByAnalysis
 				result.StatusMessage = "Waiting containers have been detected"
-				msg := "container " + container.Name + " in pod " + pod.Metadata.Name + " is in waiting state"
-				info := Info{Name: container.Name, Count: 1, Namespace: pod.Metadata.Namespace, Kind: "container", Message: msg}
-				result.Info = append(result.Info, info)
+				result.Info = append(result.Info, Info{
+					Name:      container.Name,
+					Namespace: pod.ObjectMeta.Namespace,
+					Kind:      "container", // REVIEW: why is this hard coded here while taken from data in other places?!
+					Count:     1,           // REVIEW: is Count ever going to be != 1?
+					Message:   fmt.Sprintf("container %s in pod %s is in waiting state", container.Name, pod.ObjectMeta.Name),
+				})
 			}
 		}
 	}
-
-	return result, nil
+	return result
 }
 
 // CheckEventLogForErrors checks all events in the supplied project and if any
 // are not type 'Normal' (i.e. Warning or Error), it will add them to the
 // returned results.
-func CheckEventLogForErrors(JSONResourceFactory DumpedJSONResourceFactory) (Result, error) {
-	result := Result{Status: analysisErrorNotDiscovered, StatusMessage: "this issue was not detected", CheckName: "check eventlog for any errors", Info: []Info{}, Events: []Event{}}
-	events := Events{}
-	if err := JSONResourceFactory(filepath.Join("definitions", "events.json"), &events); err != nil {
-		result.Status = analysisErrorReadingDumpedResource
-		result.StatusMessage = "Error executing task: " + err.Error()
-		return result, err
+func CheckEventLogForErrors(events types.EventList) Result {
+	result := Result{
```

Comment:
ah, I wondered if this was good practice or not! I wanted to do this formatting too as the line was so long, but wasn't sure it was good go, nice to know this is OK :)